xarray/backends/zarr.py

```diff
@@ -276,10 +276,12 @@ def open_group(
         group=None,
         consolidated=False,
         consolidate_on_close=False,
+        chunk_store=None,
     ):
         import zarr
 
-        open_kwargs = dict(mode=mode, synchronizer=synchronizer, path=group)
+        open_kwargs = dict(mode=mode, synchronizer=synchronizer, path=group,
+                           chunk_store=chunk_store)
```

Comment:
Regarding @rabernat's comment above, I think the best way forward might be to check the zarr version somewhere around here:

```python
if chunk_store is not None:
    if LooseVersion(zarr.__version__) < LooseVersion('2.4.0'):
        raise ValueError('...version not supported...')
```

Reply:
`zarr          2.3     (2019-03-25) 2.4     (2020-01-11) <`

I think we can just bump zarr to 2.4? It's almost 6 months old and so it would be in line with our "6 months for optional dependencies policy": https://xarray.pydata.org/en/stable/installing.html#minimum-dependency-versions

Reply:
> It's almost 6 months old and so it would be in line with our "6 months for optional dependencies policy":

According to [github](https://github.com/zarr-developers/zarr-python/releases), Zarr 2.4.0 was released on Apr. 28. In any case, the relevant commit (https://github.com/zarr-developers/zarr-python/commit/c18537d3695c7ad6fb582e880110749b5f761a77) was merged on Apr. 30. So I think we do need a new Zarr release.

Reply:
Or maybe we don't need a Zarr release. We might be able to test against Zarr master and use `LooseVersion(zarr.__version__) > LooseVersion('2.4.0')`.

Reply:
Indeed, our min deps script is currently broken. When I run it with https://github.com/pydata/xarray/pull/4178, I see:
`zarr          2.3     (2019-03-25) 2.3     (2019-03-25) =`

So we're supporting the right min version of zarr.

Reply:
The easiest way to do this without explicit version checks would probably be to only insert the argument it isn't the default value:
```python
        open_kwargs = dict(mode=mode, synchronizer=synchronizer, path=group)
        if chunk_size is not None:
            open_kwargs['chunk_store'] = chunk_store
```

Reply:
I made the suggested change. (Thanks @rabernat and @shoyer!)